core/base/src/main/java/alluxio/AlluxioURI.java

```diff
@@ -110,7 +110,10 @@ public AlluxioURI(String scheme, Authority authority, String path, Map<String, S
    * @param child the child
    */
   public AlluxioURI(AlluxioURI parent, AlluxioURI child) {
-    mUri = URI.Factory.create(parent.mUri, child.mUri);
+    // Allow parent to be something similar to s3://bucket_name
+    URI parentUri = parent.toString().endsWith(AlluxioURI.SEPARATOR) ? parent.mUri :
+        new AlluxioURI(parent.toString() + AlluxioURI.SEPARATOR).mUri;
```

Comment:
I'm not a huge fan of adding this to the `AlluxioURI` constructor. It adds extra string parsing because we call `new AlluxioURI` first which in turns calls another `URI.Factory.create` which can be slow.

Is there a better way we can support when the root UFS  doesn't have a trailing slash without `AlluxioURI` class?

Reply:
Changed to support in AlluxioURI, but I also feel strange becuase it also supports 
```
alluxio://host:port
hdfs://host:port
```
without trailing slash, do we also want to support those senarios?

Reply:
What does the `java.net.URI` class return for `getPath()` when we pass in `alluxio://host:port` or `hdfs://host:port`? Our URI class should be mirroring the behavior there